### PR TITLE
Fixing incorrect 500 error on get export job status.

### PR DESCRIPTION
**Pull request by v-isyamauchi-gh**
## Description
Return 400 instead of 500 for export jobs with invalid resource types

When an $export job is created with an invalid _type parameter, the orchestrator now validates resource types before enqueuing child jobs. Invalid types cause the job to fail with 400 Bad Request instead of an unhandled ResourceNotFoundException that surfaced as 500 Internal Server Error.

## Related issues
Addresses [issue #204452].

[Bug 204452](https://microsofthealth.visualstudio.com/Health/_workitems/edit/204452): Export Job returns 500 when customer enters invalid type.

## Testing
Tested by adding UTs and E2E tests.

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- When changing or adding behavior, if your code modifies the system design or changes design assumptions, please create and include an [ADR](https://github.com/microsoft/fhir-server/blob/main/docs/arch).
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/main/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/main/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs [hunk 1]
```diff
@@ -84,5 +86,19 @@ protected static ExportJobRecord CreateExportRecord(ExportJobRecord record, long
...
+            EnsureArg.IsNotNull(searchService, nameof(searchService));
+            EnsureArg.IsNotNull(record, nameof(record));
+
+            var invalidTypes = resourceTypes?.Where(t => !searchService.IsValidResourceType(t)).ToList();
+            if (invalidTypes != null && invalidTypes.Count > 0)
+            {
+                var message = $"Invalid resource type(s): {string.Join(", ", invalidTypes)}";
+                record.FailureDetails = new JobFailureDetails(message, HttpStatusCode.BadRequest);
+                throw new JobExecutionException(message, record, false);
+            }
```
[the remark this answers is not in the document]
Comment (v-isyamauchi-gh) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs]: Fixed

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportTests.cs [hunk 2]
```diff
@@ -569,5 +569,48 @@ public async Task GivenExportJobExists_WhenCancelledThenStatusCheckedThenCancell
...
+            {
+                await Task.Delay(TimeSpan.FromSeconds(2));
+
+                using HttpRequestMessage statusRequest = new HttpRequestMessage(HttpMethod.Get, contentLocation);
+                using HttpResponseMessage statusResponse = await _client.SendAsync(statusRequest);
+                statusCode = statusResponse.StatusCode;
+            }
+
+            // Step 3 — Assert the job failed with 400 Bad Request
+            Assert.Equal(HttpStatusCode.BadRequest, statusCode);
```
[the remark this answers is not in the document]
Comment (v-isyamauchi-gh) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportTests.cs:625]: By design. We need to check if the status is BadRequest, not any non-Accepted status.

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (v-isyamauchi-gh) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs]: Fixed

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs [hunk 3]
```diff
@@ -84,5 +86,22 @@ protected static ExportJobRecord CreateExportRecord(ExportJobRecord record, long
...
+
+            var invalidTypes = resourceTypes?
+                .Select(t => t?.Trim())
+                .Where(t => string.IsNullOrEmpty(t) || !searchService.IsValidResourceType(t))
+                .ToList();
+            if (invalidTypes != null && invalidTypes.Count > 0)
+            {
+                var message = $"Invalid resource type(s): {string.Join(", ", invalidTypes)}";
+                record.FailureDetails = new JobFailureDetails(message, HttpStatusCode.BadRequest);
+                throw new JobExecutionException(message, record, false);
```
[the remark this answers is not in the document]
Comment (v-isyamauchi-gh) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs]: Fixed

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportTests.cs [hunk 4]
```diff
@@ -569,5 +574,62 @@ public async Task GivenExportJobExists_WhenCancelledThenStatusCheckedThenCancell
...
+                statusCode = statusResponse.StatusCode;
+                responseContent = statusResponse.Content != null ? await statusResponse.Content.ReadAsStringAsync() : null;
+            }
+
+            // Step 3 — Assert the job failed with 400 Bad Request
+            Assert.Equal(HttpStatusCode.BadRequest, statusCode);
+
+            // Step 3a - Make sure the response content is not empty and contains an OperationOutcome with the expected invalid resource type
+            var resource = _fhirJsonParser.Parse<Resource>(responseContent);
+            var operationOutcome = Assert.IsType<OperationOutcome>(resource);
```
[the remark this answers is not in the document]
Comment (v-isyamauchi-gh) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportTests.cs:630]: Fixed

File: src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportOrchestratorJobTests.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+            var record = CreateRecord("SpecimenPerson");
+            var resourceTypes = new[] { "SpecimenPerson" };
+
+            var ex = Assert.Throws<JobExecutionException>(
+                () => TestableExportOrchestratorJob.InvokeValidateResourceTypes(_searchService, resourceTypes, record));
+
+            Assert.Contains("SpecimenPerson", ex.Message);
+            Assert.Equal(HttpStatusCode.BadRequest, record.FailureDetails.FailureStatusCode);
+        }
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportOrchestratorJobTests.cs:122]: Can you add a test to cover the case of a search parameter name that only contains whitespace returning '\<empty\>'?

Comment (v-isyamauchi-gh) [reply src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Export/ExportOrchestratorJobTests.cs:122]: Fixed by adding a new case for a resource type with whitespace.

**Review by LTA-Thinking — APPROVED** [review]
